src/main/java/online/book/store/validation/FieldMatchValidator.java

```diff
@@ -0,0 +1,18 @@
+package online.book.store.validation;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import online.book.store.dto.UserRegistrationRequestDto;
+
+public class FieldMatchValidator implements
+        ConstraintValidator<FieldMatch,
+        UserRegistrationRequestDto> {
+
+    @Override
+    public boolean isValid(
+            UserRegistrationRequestDto requestDto,
+            ConstraintValidatorContext constraintValidatorContext
+    ) {
+        return requestDto.getPassword().equals(requestDto.getRepeatPassword());
```

Comment:
Can getPassword() return null?
If it will return null, how are we going to handle NPE?
Just use [Objects#equals](https://www.baeldung.com/java-comparing-objects#equals-static).